test/jdk/javax/swing/ImageIcon/ImageIconTest.java

```diff
@@ -0,0 +1,121 @@
+                System.out.println("Testing for ArgType " + a + " for case " + v);
+                boolean expected = true;
+                boolean passed = false;
+                try {
+                    switch (a) {
+                       case FILE :
+                           expected = false;
+                           String s = (v == ArgVal.NULL) ? null : fileName;
+                           new ImageIcon(s);
+                           passed = true; // no exception expected for this case
+                           break;
+                       case URL :
+                           if (v == ArgVal.NULL) {
+                               new ImageIcon((URL)null);
+                           } else if (v == ArgVal.INVALIDDATA) {
+                               expected = false;
+                               new ImageIcon("file://" + imgName, "gif");
+                               passed = true; // no exception expected for this case
+                           }
+                           break;
+                       case BYTEARRAY :
+                           if (v == ArgVal.NULL) {
+                               byte[] bytes = null;
+                               new ImageIcon(bytes);
+                           } else if (v == ArgVal.INVALIDDATA) {
+                               expected = false;
+                               new ImageIcon(new byte[0], "gif");
+                               passed = true; // no exception expected for this case
+                           }
+                           break;
+                       case IMAGE :
+                           if (v == ArgVal.NULL) {
+                               new ImageIcon((Image)null);
+                           } else if (v == ArgVal.INVALIDDATA) {
+                               expected = false;
+                               new ImageIcon((Image)Toolkit.getDefaultToolkit().createImage(fileName), "gif");
+                               passed = true; // no exception expected for this case
+                           }
+                           break;
+                    }
+                } catch (NullPointerException e) {
+                    if (expected) {
+                        passed = true;
+                    }
+                }
+                if (expected && !passed) {
+                   System.err.println("Did not receive expected exception for : " + a);
+                   throw new RuntimeException("Test failed");
+                }
+                if (!expected && !passed) {
+                   System.err.println("Received unexpected exception for : " + a);
+                   throw new RuntimeException("Test failed");
+                }
+            }
+        }
+
+        // test setter
+        try {
+            ImageIcon ii = new ImageIcon();
+            ii.setImage((Image)null);
+            throw new RuntimeException("No NPE");
+        } catch (NullPointerException e) {
```

Comment:
It occurred to me that you could include this in the switch with a new ARGTYPE of "SETIMAGE".
Bit of a misnomer as its the type + the method but then you can also check the non-null invalid data type.